README.md

```diff
@@ -1 +1,151 @@
+* CMake (version >= 3.12)
+* Boost (version >= 1.63)
+* curl and its development libraries
+
+## Building
+```
+# fetch third-party dependencies
+git submodule init
+git submodule update
+
+# compile and build
+mkdir -p build
+cd build
+cmake ..
+make -j 2
+
+# to run the example
+./src/wavefront-sdk
+```
+
+## Usage
+
+### Create Wavefront Client
+
+#### Create a `WavefrontProxyClient`
+
+Assume you have a running Wavefront proxy listening on at least one of metrics / direct-distribution / tracing ports and you know the proxy hostname.
+
+```cpp
+#include "proxy/WavefrontProxyClient.h"
+/**
+* Create Proxy Client
+* host: Hostname of the Wavefront proxy (required)
+* metrics_port: Metrics Port on which the Wavefront proxy is listening on
+* distribution_port: Distribution Port on which the Wavefront proxy is listening on
+* tracing_port: Tracing Port on which the Wavefront proxy is listening on
+*/
+
+ WavefrontProxyClient::Builder builder(<host>);
+ WavefrontProxyClient *client = builder.setMetricsPort(2878).
+                                        setDistributionPort(40000).
+                                        setTracingPort(30000).
+                                        build();
+```
+
+#### Create a `WavefrontDirectIngestionClient`
+
+Assume you have a running Wavefront cluster and you know the server URL (example - https://mydomain.wavefront.com) and the API token.
+
+```cpp
+#include "direct_ingestion/WavefrontDirectIngestionClient.h"
+/** 
+* Create Direct Ingestion Client
+* server: Server address, Example: https://mydomain.wavefront.com (required)
+* token: Token with Direct Data Ingestion permission granted (required)
+* max_queue_size: Max Queue Size, size of internal data buffer for each data type. 50000 by default
+* batch_size: Batch Size, amount of data sent by one API call, 10000 by default
+* flush_interval_seconds: Buffer flushing interval time, 5 seconds by default
+*/
+
+WavefrontDirectIngestionClient::Builder builder1(<server>, <token>);
+WavefrontDirectIngestionClient *client1 = builder1.build();
+
+// must be explicitly called, start flushing thread
+client1->start();
+```
+
+### Send Data Point
+
+Using following functions to send data point to Wavefront via Proxy or Direct ingestion.  
+The differences between sending data via Proxy and Direct ingestion are:
+* Direct ingestion calls `report` API and usually sends data in batch and compressed format 
+```cpp
+// 1) Send Metric to Wavefront
+  /*
+   * Wavefront Metrics Data format
+   * <metricName> <metricValue> [<timestamp>] source=<source> [pointTags]
+   *
+   * Example: "new-york.power.usage 42422 1533529977 source=localhost datacenter=dc1"
+   */
+  client->sendMetric("new-york.power.usage", 42422.0, 1533529977L,
+        "localhost", {{"datacenter", "dc1"}});
+
+  // 2) Send Direct Distribution (Histogram) to Wavefront
```

Comment:
You should also have sample code for 
// 2) Send Delta Counter to Wavefront     

see the README for https://github.com/wavefrontHQ/wavefront-java-sdk